src/v0/destinations/marketo_bulk_upload/util.js

```diff
@@ -38,44 +43,52 @@ const getAccessToken = async config => {
     ) {
       return resp.response.data.access_token;
     }
-    throw new CustomError("Could not retrieve authorisation token", 400);
+    throw new NetworkInstrumentationError(
+      "Could not retrieve authorisation token"
+    );
   }
   if (resp.response) {
     // handle for abortable codes
     if (
       ABORTABLE_CODES.indexOf(resp.response.code) > -1 ||
       (resp.response.code >= 400 && resp.response.code <= 499)
     ) {
-      throw new CustomError(resp.response.code, 400);
+      throw new AbortedError(resp.response.code, 400, resp);
     } // handle for retryable codes
     else if (RETRYABLE_CODES.indexOf(resp.response.code) > -1) {
-      throw new CustomError(resp.response.code, 500);
+      throw new RetryableError(resp.response.code, 500, resp);
     } // handle for abortable codes
     else if (resp.response.response) {
       if (ABORTABLE_CODES.indexOf(resp.response.response.status) > -1) {
-        throw new CustomError(
+        throw new AbortedError(
           resp.response.response.statusText ||
             "Error during fetching access token",
-          400
+          400,
+          resp
         );
       } // handle for throttled codes
       else if (THROTTLED_CODES.indexOf(resp.response.response.status) > -1) {
-        throw new CustomError(
+        throw new ThrottledError(
           resp.response.response.statusText ||
             "Error during fetching access token",
-          500
+          resp
         );
       }
       // Assuming none we should retry the remaining errors
-      throw new CustomError(
+      throw new RetryableError(
         resp.response.response.statusText ||
           "Error during fetching access token",
-        500
+        500,
+        resp
       );
     }
-    throw new CustomError("Could not retrieve authorisation token", 400);
+    throw new NetworkInstrumentationError(
+      "Could not retrieve authorization token"
+    );
   }
-  throw new CustomError("Could not retrieve authorisation token", 400);
+  throw new NetworkInstrumentationError(
+    "Could not retrieve authorization token"
+  );
```

Comment:
Auth failures here Auth errors should be here @saikumarrs can you confirm ?

Reply:
Yes, agreed.